tests/TODO.md

```diff
@@ -186,6 +186,20 @@ groups.
 * **Effective space**. Veryfy calculations of `get_free_space()` and
   `get_effective_space()`. Test cases are the same as for Couple's test
   **Effective space**. **TODO**: docs.
+* **History**. Verify applying of `GroupHistoryEntry` coming from history
+  database. If some of group backends are absent in history entry, they
+  must be removed from Group's set of backends. Actions are performed in
+  `Group::apply(const GroupHistoryEntry &)`.
+  1. *No changes #1*. Apply entry with the same set of backends. Nothing must
+  change.
+  2. *No changes #2*. Apply entry with the same set of backends plus new
+  backends. Nothing must change.
+  3. *One backend removed*. Apply entry with one backend absent. It must
+  disappear from the list of Group's backneds.
+  4. *All backends removed #1*. Apply entry with empty set of backends.
+  List of Group's backends must become empty.
+  5. *All backends removed #2*. Apply entry with totally different set
+  of backends. List of Group's backends must become empty.
 
```

Comment:
You can add a test case for 'automatic' entry type to ensure that backends from this entry do not influence the list of group's backends.